app/scripts/services/payment.js

```diff
@@ -1,16 +1,56 @@
 'use strict';
 
 angular.module('confRegistrationWebApp')
-  .factory('payment', function ($q, $http, $filter, error) {
+  .factory('payment', function ($q, $http, $filter, cruPayments, envService, error, gettextCatalog) {
+    // Load the TSYS manifest
+    // Returns a promise that resolves to an object containing the manifest and device id
+    function loadTsysManifest (payment) {
+      var url = 'payments/appManifest';
+      return $http.get(url, { data: payment }).then(function (res) {
+        return res.data;
+      });
+    }
+
     // Modify a credit card payment to use a tokenized credit card instead of real credit card data
-    function tokenizeCreditCardPayment (payment) {
+    // Tokenize the credit card via Authorize.NET
+    function tokenizeCreditCardPaymentAuthorizeNet (payment) {
       return $http.get('payments/ccp-client-encryption-key').then(function (res) {
         var ccpClientEncryptionKey = res.data;
         ccp.initialize(ccpClientEncryptionKey);
         payment.creditCard.lastFourDigits = ccp.getAbbreviatedNumber(payment.creditCard.number);
         payment.creditCard.number = ccp.encrypt(payment.creditCard.number);
```

Comment:
@canac You're still encrypting the token? Is that necessary? I'm finally getting to implement this with the give site. I wasn't sure if the cvv still needed to be encrypted either. If it does, that might be nice to move into the cru-payments library.

@hlbraddock Why does the CVV need to be passed to TSYS during tokenization if the backend still needs it for a realtime transaction? If it is never needed during tokenization, that might make https://github.com/CruPCI/cru-payments/pull/16 cleaner. Is it really needed during both tokenization and when interacting with TSYS on the backend?